modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py

```diff
@@ -36,30 +44,37 @@ class PandasOnRayDataframePartition(PandasDataframePartition):
 
     Parameters
     ----------
-    data : ray.ObjectRef
-        A reference to ``pandas.DataFrame`` that need to be wrapped with this class.
-    length : ray.ObjectRef or int, optional
+    data : ObjectIDType or DeferredExecution
+        A reference to ``pandas.DataFrame`` that needs to be wrapped with this class
+        or a reference to DeferredExecution that needs to be executed on demand.
+    length : ObjectIDType or int, optional
         Length or reference to it of wrapped ``pandas.DataFrame``.
-    width : ray.ObjectRef or int, optional
+    width : ObjectIDType or int, optional
         Width or reference to it of wrapped ``pandas.DataFrame``.
-    ip : ray.ObjectRef or str, optional
+    ip : ObjectIDType or str, optional
         Node IP address or reference to it that holds wrapped ``pandas.DataFrame``.
-    call_queue : list
-        Call queue that needs to be executed on wrapped ``pandas.DataFrame``.
     """
 
     execution_wrapper = RayWrapper
 
-    def __init__(self, data, length=None, width=None, ip=None, call_queue=None):
+    def __init__(
+        self,
+        data: Union[ray.ObjectRef, "ClientObjectRef", DeferredExecution],
+        length: int = None,
+        width: int = None,
+        ip: str = None,
+    ):
         super().__init__()
-        assert isinstance(data, ObjectIDType)
-        self._data = data
-        if call_queue is None:
-            call_queue = []
-        self.call_queue = call_queue
-        self._length_cache = length
-        self._width_cache = width
-        self._ip_cache = ip
+        if isinstance(data, DeferredExecution):
+            data.ref_count(1)
+        self._data_ref = data
+        # The metadata is stored in the MetaList at 0 offset. If the data is
+        # a DeferredExecution, the _meta will be replaced with the list, returned
+        # by the remote function. The returned list may contain data for multiple
+        # results and, in this case, _meta_offset corresponds to the meta related to
+        # this partition.
+        self._meta = MetaList([length, width, ip])
+        self._meta_offset = 0
```

Comment:
Why should one partition contain meta information for several results?

Reply:
In case of a batch execution, the meta is returned for all the results in a single list. Materialization of this list is also don in a single batch operation. Multiple partitions refer to a single list. When it's materialized, it's materialized for all the partitions at once.